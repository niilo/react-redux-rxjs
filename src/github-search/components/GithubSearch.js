import React from 'react'

import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'
import {queryData} from '../githubSearchActions'

const GithubSearchListItem = ({item}) => (
  <li>id: {item.id}, Login: {item.login}</li>
)

const GithubSearchQuery = ({query}) => <input type='text' onChange={query} />

const GithubSearchList = ({result}) => {
  if (result.isFetching) return <p>Loading</p>

  if (result.error) return <p>Error</p>

  return (
    <ul>
      {result.data && result.data.items
        ? result.data.items.map(account => (
            <GithubSearchListItem key={account.id} item={account} />
          ))
        : null}
    </ul>
  )
}

const GithubSearch = props => {
  return (
    <div className='container'>
      <GithubSearchQuery query={props.queryData} />
      <GithubSearchList result={props.appData} />
    </div>
  )
}

const mapStateToProps = state => ({
  appData: state.githubSearchReducer
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      queryData
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(GithubSearch)
