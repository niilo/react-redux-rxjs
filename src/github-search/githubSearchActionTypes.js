export const GITHUB_QUERY = 'github/QUERY'
export const GITHUB_CANCEL_QUERY = 'github/CANCEL_QUERY'
export const GITHUB_QUERY_FULFILLED = 'github/QUERY_FULFILLED'
export const GITHUB_QUERY_REJECTED = 'github/QUERY_REJECTED'
