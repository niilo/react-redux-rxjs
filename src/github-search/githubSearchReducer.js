import {
  GITHUB_QUERY,
  GITHUB_CANCEL_QUERY,
  GITHUB_QUERY_FULFILLED,
  GITHUB_QUERY_REJECTED
} from './githubSearchActionTypes'

const initialState = {
  data: [],
  query: '',
  dataFetched: false,
  isFetching: false,
  error: false
}

export default function githubSearchReducer (state = initialState, action) {
  switch (action.type) {
    case GITHUB_QUERY:
      return Object.assign({}, ...state, {
        query: action.query,
        isFetching: true
      })

    case GITHUB_QUERY_FULFILLED:
      return Object.assign(
        {},
        ...state,
        {data: action.payload.response},
        {isFetching: false}
      )

    case GITHUB_QUERY_REJECTED:
      return Object.assign({}, ...state, {error: true}, {isFetching: false})

    default:
      return state
  }
}
