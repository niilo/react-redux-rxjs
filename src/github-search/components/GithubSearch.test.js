import React from 'react'
import renderer from 'react-test-renderer'
import {GithubSearch} from './GithubSearch'
import githubSearchReducer from '../githubSearchReducer'
import {GITHUB_QUERY_FULFILLED} from '../githubSearchActionTypes'
import mockSearchResponse from './__mockData__/github-search-result.json'

it('github search result list renders correctly', () => {
  const action = {
    type: GITHUB_QUERY_FULFILLED,
    payload: {
      response: mockSearchResponse
    }
  }
  const state = githubSearchReducer(undefined, action)
  const data = {
    appData: state
  }
  const tree = renderer.create(<GithubSearch {...data} />).toJSON()
  expect(tree).toMatchSnapshot()
})
