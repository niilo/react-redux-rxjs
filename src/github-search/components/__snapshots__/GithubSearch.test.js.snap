// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`github search result list renders correctly 1`] = `
<div
  className="container"
>
  <input
    onChange={undefined}
    type="text"
  />
  <ul>
    <li>
      id: 
      681559
      , Login: 
      niilo
    </li>
    <li>
      id: 
      4352669
      , Login: 
      Kargamborgh
    </li>
    <li>
      id: 
      4191136
      , Login: 
      Purrnuu
    </li>
  </ul>
</div>
`;
